builder/azure/arm/template_factory.go

```diff
@@ -10,15 +11,33 @@ import (
 
 	"github.com/hashicorp/packer-plugin-azure/builder/azure/common/constants"
 	"github.com/hashicorp/packer-plugin-azure/builder/azure/common/template"
+	"golang.org/x/crypto/ssh"
 )
 
 type templateFactoryFunc func(*Config) (*resources.Deployment, error)
 
-func GetKeyVaultDeployment(config *Config) (*resources.Deployment, error) {
+func GetWinRMKeyVaultDeployment(config *Config) (*resources.Deployment, error) {
+	return GetKeyVaultDeployment(config, config.winrmCertificate)
+}
+
+func GetSSHKeyVaultDeployment(config *Config) (*resources.Deployment, error) {
+	privateKey, err := ssh.ParseRawPrivateKey(config.Comm.SSHPrivateKey)
+	if err != nil {
+		return nil, err.(error)
+	}
+	pk, _ := privateKey.(*rsa.PrivateKey)
```

Comment:
This one will require a bit of 🤔.

Reply:
This effectively re-uses the certificate logic for when the plugin needs to create its own certificate.
I'm not that knowledgable in that space, so I'm somewhat hesitant to try and support it all.

I wonder if the best path forward here is to either:
a) throw a (validation) error if a custom cert is supplied saying "not supported (yet)"
b) do a type check and throw a (validation) error if the cert is not an RSA key

Reply:
+1 on @JenGoldstrich's suggestion then, if RSA only is supported, it's fair to throw an error if the key isn't in the expected format.

Reply:
I've pushed a change for this, though I haven't tested it yet (yolo!)
I'm intending to test it out over the weekend.

Reply:
Okay, just tested it out - looks good.
(fyi: I tested it via https://github.com/hashicorp/packer-plugin-azure/pull/232, rather than independently, so I didn't have to worry about setting up pre-existing base images with ssh keys etc)